### ISSUE TITLE
IRIS: IOC for McLennan PM600 Motion Controller

### ISSUE THREAD

**Issue by kjwoodsISIS** [ISISComputingGroup/IBEX#1099]
As a scientist, I need an IOC for the McLennan PM600 Motion Controller, so that I can control the IRIS rotating centre-stick.

##### Acceptance Criteria
1. I can control a McLennan PM600 Motion Controller a console (using caput, caget, etc.). 
2. I can control a McLennan PM600 Motion Controller from a genie_python script. 

##### Notes
1. There is a [PM600 driver](http://www.aps.anl.gov/epics/modules/manufacturer.php#McLennan%20Servo%20Supplies) available from the EPICS web-site.
1. [McLennan web-site](http://www.mclennan.co.uk/)

**Comment by FreddieAkeroyd** [comment]
Code pushed in ISISComputingGroup/EPICS-ioc#60

**Comment by AdrianPotter** [comment]
Some notes from testing the IOC this week. I've been using the test stage David was working with. The IOC talks to the controller as expected, but the controller isn't getting the motor to move. I couldn't determine why.

The IOC is good, but there are a few changes to make it work as expected with IBEX:

- The PV names are currently of the form `MOT:MTR01`. They are missing their controller index (e.g. `MTR0101`). To avoid overlap with the Galils, the macros need setting up so that the McLennan controllers occupy a different band of controller numbers. The Galils currently use the defined address macros to assign controller numbers. Do we want to do the same with the McLennans?
- The description for the Galil PV should have `(GALIL)` appended as in `(MCLEN)` to distinguish the motor types. I've already created a branch of the ibex_gui which widens the motor record OPI to accommodate the extra length.
- The Galils only pass the `MTR*` part of the macro to the db, the McLennan passes `MOT:MTR*`. The most obvious effect of this is that the PV description for the McLennan motor record contains a `MOT:` that the Galils don't. This at least should be corrected.

**Comment by AdrianPotter** [comment]
https://github.com/ISISComputingGroup/EPICS-motor/pull/1
https://github.com/ISISComputingGroup/EPICS-ioc/pull/68

Note that the IOC PR are changes I've made since David's code was added to the repo. The ticket involves the entirety of ioc/master/MCLEN, not just the changes I've made.

Also note that, unlike the Galil, this is a level 1 motor and supports very little simulation functionality. Rather than spending more time trying to emulate level 3 simulation capability, I've returned the ticket as is owing to the urgency of getting the McLennan ready for Iris. Enhancing simulation behaviour can be submitted under a separate ticket if required.
